src/App.js

```diff
@@ -0,0 +1,42 @@
+import { Route, Routes, useLocation } from 'react-router-dom';
+import './App.css';
+import { useDispatch, useSelector } from 'react-redux';
+import { useEffect } from 'react';
+
+import { fetchMissions } from './slices/MessionSlice';
+import MyProfile from './components/MyProfile';
+import Rockets from './components/Rockets';
+import NavBar from './components/NavBar';
+import MessionList from './components/MessionList';
+import { getRockets } from './slices/RocketSlice';
+
+function App() {
+  const dispatch = useDispatch();
+  const location = useLocation();
+
+  useEffect(() => {
+    dispatch(getRockets());
+  }, []);
+
+  const { messions } = useSelector((state) => state.missions);
+  useEffect(() => {
+    if (messions.length <= 2 && location.pathname === '/missions') {
+      dispatch(fetchMissions());
+    }
+  }, [location.pathname]);
+
```

Comment:
- [ ] @team great work implementing the changes requested however when running the app the rocket page is not loading as shown in the video in the link below also has an error as shown in the image below

![image](https://user-images.githubusercontent.com/61408860/208974581-d8ff16f2-9769-4b81-83bb-aef6e09b4126.png)

https://www.loom.com/share/349b5151b47644dfa16e99100b5c473a

Reply:
Hi @danmainah ,  thanks  for the feedback .
Did make sure to add the .env file and variables as noted in the pull request description?